src/minifluxapi.cpp

```diff
@@ -139,11 +139,38 @@ bool MinifluxApi::mark_article_read(const std::string& guid, bool read)
 	return true;
 }
 
-bool MinifluxApi::update_article_flags(const std::string& /* oldflags */,
-	const std::string& /* newflags */,
-	const std::string& /* guid */)
+bool MinifluxApi::star_article(const std::string& guid, bool star)
 {
-	return false;
+	json args;
+	std::string putcontent;
+	putcontent = strprintf::fmt(
+				"/v1/entries/%s/bookmark",
+				guid);
+	
+	// this function is actually a toggle, so we do not need the star parameter
```

Comment:
This is pretty bad. If one stars an item in Newsboat, then un-stars it in Miniflux, then notices it's still starred in Newsboat and un-stars it there, the item will end up starred again. Is there an API endpoint to check if an items is already starred?

Reply:
You can get details about any feed entry [Get Entry](https://miniflux.app/docs/api.html#endpoint-get-entry), which includes the starred state. I did it this way because I thought #1130 implied `newsboat` was not able to save states in feed entries. If that can be done while downloading the feed entries let me know how (where to store this entry state) and I can implement it.

Reply:
> I thought https://github.com/newsboat/newsboat/issues/1130 implied newsboat was not able to save states in feed entries.

Correct, it's (currently) unable to do that. So what I'm suggesting is to do the check right here: get entry, check its starred state, then toggle it if needed. That means we'll do 2 queries, which is slower than 1, but at least we avoid the problem I described above.

We still run a chance that a second client is toggling the flag at the same time, and we end up negating each other's actions. But that's a shortcoming of the API and nothing that Newsboat can fix.

Reply:
Just applied that change. Hopefully in the future `newsboat` allows to do this in a proper way.